tests/_support/Step/Acceptance/OAuthClientsStep.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * @package     redCORE
+ * @subpackage  Step
+ * @copyright   Copyright (C) 2008 - 2019 redCOMPONENT.com. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+namespace Step\Acceptance;
+use Page\OAuthClientsPage;
+
+/**
+ * Class OAuthClientsStep
+ * @package Step\Acceptance
+ */
+class OAuthClientsStep extends AbstractStep
+{
+	/**
+	 * @param $clientID
+	 * @param $redirectURI
+	 * @throws \Exception
+	 */
+	public function createNewOAuthClient($clientID,$redirectURI)
+	{
+		$I = $this;
+		$I->amOnPage(OAuthClientsPage::$URL);
+		$I->waitForText(OAuthClientsPage::$titleOAuth, 30, OAuthClientsPage::$h1);
+		$I->click(OAuthClientsPage::$buttonNew);
+		$I->waitForText(OAuthClientsPage::$titleOAuthClient, 30, OAuthClientsPage::$h1);
+		$I->fillField(OAuthClientsPage::$fieldClientID, $clientID);
+		$I->fillField(OAuthClientsPage::$fieldURI, $redirectURI);
+		$I->click(OAuthClientsPage::$buttonSaveClose);
+	}
+
+	/**
+	 * @throws \Exception
+	 */
+	public function waitForSuccess()
+	{
+		$I = $this;
+		$I->waitForText(OAuthClientsPage::$messageSuccess, 30, OAuthClientsPage::$messageContainer);
+		$I->waitForText(OAuthClientsPage::$messageSaveSuccess, 30, OAuthClientsPage::$messageContainer);
+	}
+
+	/**
+	 * @param $clientID
+	 */
+	public function searchOAuthClient($clientID)
+	{
+		$I= $this;
+		$I->fillField(OAuthClientsPage::$fieldSearch, $clientID);
+		$I->click(OAuthClientsPage::$search);
+	}
+
+	/**
+	 * @param $clientID
+	 * @param $clientID2
+	 * @throws \Exception
+	 */
+	public function editOAuthClient($clientID,$clientID2)
+	{
+		$I = $this;
+		$I->amOnPage(OAuthClientsPage::$URL);
+		$I->waitForText(OAuthClientsPage::$titleOAuth, 30, OAuthClientsPage::$h1);
+		$I->searchOAuthClient($clientID);
+		$I->see($clientID, OAuthClientsPage::$oauthClientsList);
```

Comment:
Please change see to waitForText

Reply:
And why should that be done?, You can add another waitForText..but it should not replace `see`, one is an implicit wait while the other is an assertion and that's what the test is supposed to do.

Reply:
```
I->search
I->waitForText
I->see
```

Reply:
I think we should be waitForText before when use $I->see to avoid fail random

Reply:
I think that is what I mentioned here? https://github.com/redCOMPONENT-COM/redCORE/pull/868#discussion_r284191002

Reply:
Yes @puneet0191. Thanks